tests/trainers/java/files_and_directories.java

```diff
@@ -0,0 +1,185 @@
+/**
+Test to check the Java File module.
+TODO
+*/
+import java.io.*;
+import java.io.IOException;
+import java.util.UUID;
+
+class Test {
+    int step = 1;
+    String textContent = "Content of the text file";
+    int errors = 0;
+
+    private String createTextFile(String filepath) {
+        try {
+            FileWriter file = new FileWriter(filepath);
+            file.write(this.textContent);
+            file.close();
+            return "OK";
+        } catch (IOException ex) {
+            return "KO - " + ex.getMessage();
+        }
+    }
+
+    private String readTextFile(String filepath) {
+        try {
+            FileReader file = new FileReader(filepath);
+            char [] a = new char[50];
+            file.read(a);
+            String output = "";
+            for(char c : a)
+                output = output + c;
+            file.close();
+            output = output.trim(); //Note: there are some hidden chars at the end has to be removed
+
+            if (output.equals(this.textContent))
+                return "OK";
+            else
+                return "KO - file content doesn't match with '" + this.textContent + "'";
+        } catch (IOException ex) {
+            return "KO - " + ex.getMessage();
+        }
+    }
+
+    private String createBinaryFile(String inputFile, String outputFile) {
+        try (
```

Comment:
You're missing the `close` for the `inputStream` and the `outputStream` in the whole block.

Reply:
done, thanks!